Tests/RegexTests/MatchTests.swift

```diff
@@ -14,22 +14,32 @@ import XCTest
 @testable import _StringProcessing
 
 struct MatchError: Error {
-    var message: String
-    init(_ message: String) {
-        self.message = message
-    }
+  var message: String
+  init(_ message: String) {
+    self.message = message
+  }
 }
 
 func _firstMatch(
   _ regexStr: String,
   input: String,
+  validate: Bool,
   syntax: SyntaxOptions = .traditional
 ) throws -> (String, [String?]) {
   let regex = try Regex(regexStr, syntax: syntax)
   guard let result = try regex.firstMatch(in: input) else {
     throw MatchError("match not found for \(regexStr) in \(input)")
   }
   let caps = result.output.slices(from: input)
+  
+  if validate {
+    var unoptRegex = try Regex(regexStr, syntax: syntax)
+    unoptRegex._setCompilerOptionsForTesting(.unoptimized)
+    guard let unoptResult = try unoptRegex.firstMatch(in: input) else {
+      throw MatchError("match not found for unoptimized \(regexStr) in \(input)")
+    }
+    XCTAssertEqual(String(input[result.range]), String(input[unoptResult.range]))
```

Comment:
Can you add an optimization bug (locally) to make sure this gets triggered? (or vice versa)

Reply:
Yep and it was triggered as expected